packages/chat-core/src/index.ts

```diff
@@ -587,6 +584,7 @@ export function promisify<T>(
     return await new Promise((resolve, reject) => {
       const timeoutId = setTimeout(() => {
         reject(new Error("The request timed out."));
+        convo.unsubscribe(subscription);
```

Comment:
It's ok if `subscription` is defined below right? On account of the callback?

Reply:
shouldl be fine 

test code

```
const a = () => {
    console.log(b);
}
const b = 'foo';
a();
```

https://www.typescriptlang.org/play?#code/MYewdgzgLgBAhjAvDAFASiQPhgbwFAyEyiQgA2ApgHRkgDmKARmgNx4C+eJ0MjSMAcgBmIEALZx0LIA